lib/galaxy/workflow/run.py

```diff
@@ -251,8 +251,15 @@ def invoke(self) -> Dict[int, Any]:
                 step_delayed = delayed_steps = True
                 self.progress.mark_step_outputs_delayed(step, why=de.why)
             except Exception as e:
-                log.exception(
-                    "Failed to schedule %s, problem occurred on %s.",
+                log_function = log.exception
+                if isinstance(e, modules.FailWorkflowEvaluation) and e.why.reason not in (
+                    "unexpected_failure",
+                    "expression_evaluation_failed",
+                ):
+                    log_function = log.info
```

Comment:
Can we maybe use an allowlist approach for when to switch to `log.info`? E.g. `e.why.reason in ("dataset_failed", "collection_failed", ...)`?
So that if we add more strings to `FailureReason` they are by default logged at exception level.

Reply:
sure, sounds good.